h2o-algos/src/main/java/hex/anovaglm/AnovaGLMUtils.java

```diff
@@ -0,0 +1,454 @@
+import static hex.anovaglm.AnovaGLMModel.AnovaGLMParameters;
+import static hex.gam.MatrixFrameUtils.GAMModelUtils.copyTwoDimTable;
+import static hex.gam.MatrixFrameUtils.GAMModelUtils.genCoefficientTable;
+import static hex.gam.MatrixFrameUtils.GamUtils.setParamField;
+import static hex.glm.GLMModel.GLMParameters;
+import static hex.glm.GLMModel.GLMParameters.Family.*;
+
+public class AnovaGLMUtils {
+  /***
+   * This method will extract the individual predictor names that will be used to build the GLM models.
+   * 
+   * @param dinfo: DataInfo generated from dataset with all predictors
+   * @param numOfPredictors: number of individual predictors
+   * @return: copy of individual predictor names in a String array.
+   */
+  public static String[] extractPredNames(DataInfo dinfo,  int numOfPredictors) {
+    String[] predNames = new String[numOfPredictors];
+    String[] frameNames = dinfo._adaptedFrame.names();
+    System.arraycopy(frameNames, 0, predNames, 0, numOfPredictors);
+    return predNames;
+  }
+
+  /**
+   * In order to calculate Type III SS, we need the individual predictors and their interactions.  For details, refer
+   * to AnovaGLMTutorial section IV
+   * 
+   * @param predNamesIndividual: string containing individual predictor names
+   * @param maxPredInt: maximum number of predictors allowed in interaction term generation
+   * @return String array with double indices.  First index refers to the predictor number, second index refers to
+   *         the names of predictors involved in generating the interaction terms.  For terms involving only
+   *         individual predictors, there is only one predictor name.
+   */
+  public static String[][] generatePredictorCombos(String[] predNamesIndividual, int maxPredInt) {
+    List<String[]> predCombo = new ArrayList<>();
+    addIndividualPred(predNamesIndividual, predCombo);  // add individual predictors
+    for (int index = 2; index <= maxPredInt; index++) {
+      generateOneCombo(predNamesIndividual, index, predCombo);
+    }
+    return predCombo.toArray(new String[0][0]);
+  }
+  
+  public static void addIndividualPred(String[] predNames, List<String[]> predCombo) {
+    int numPred = predNames.length;
+    for (int index=0; index < numPred; index++) {
+      predCombo.add(new String[]{predNames[index]});
+    }
+  }
+  
+  public static void generateOneCombo(String[] predNames, int numInteract, List<String[]> predCombo) {
+    int predNum = predNames.length;
+    int[] predInd = IntStream.range(0, numInteract).toArray();
+    int zeroBound = predNum-numInteract;
+    int[] bounds = IntStream.range(zeroBound, predNum).toArray();
+    int numCombo = hex.genmodel.utils.MathUtils.combinatorial(predNum, numInteract);
+    for (int index = 0; index < numCombo; index++) {
+      predCombo.add(predCombo(predNames, predInd));
+      if (!updatePredCombo(predInd, bounds))
+        break;  // done
+    }
+  }
+  
+  public static int find(String[][] totArray, String ele) {
```

Comment:
the function name is generic but behavior is specific - at least make this function private

Reply:
I changed it to private.